### PR TITLE
Ignore file locks on OSX NFS mounts

### PR DESCRIPTION
**Pull request by alexcrichton**
We already ignore NFS on Linux so let's do the same on OSX as well.

### CODE REVIEW

File: src/cargo/util/flock.rs [hunk 1]
```diff
@@ -267,6 +269,13 @@ fn acquire(config: &Config,
 
     match try() {
         Ok(()) => return Ok(()),
+
+        // Like above, where we ignore file locking on NFS mounts on Linux, we
+        // do the same on OSX here. Note that ENOTSUP is an OSX_specific
+        // constant.
+        #[cfg(target_os = "macos")]
+        Err(ref e) if e.raw_os_error() == Some(libc::ENOTSUP) => return Ok(),
```
Comment (Kerosene2000) [inline src/cargo/util/flock.rs]:
```rust
Ok(()) // no ?
```

Comment (alexcrichton) [inline src/cargo/util/flock.rs]: Oops!

**Pull request merged**